### PR TITLE
Feat: Administrative Adjustments and Initiative Agreements

### PR DESCRIPTION
**Pull request by AlexZorkin**
- Added backend api views, services, repo, schema for AdminAdjustment and Initiative Agreement
- Frontend AddEditViewTransaction component expanded for api data integration and state management
- ButtonConfigs for all status' setup
- Seeders and model update with a migration

### CODE REVIEW

File: backend/lcfs/db/migrations/versions/2024-05-13-15-36_0985f23b0f0c.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+Create Date: 2024-05-13 15:36:40.362106
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "0985f23b0f0c"
+down_revision = "6e08afd00978"
```
Comment (prv-proton) [inline backend/lcfs/db/migrations/versions/2024-05-13-15-36_0985f23b0f0c.py:14]: Seems to have conflict with head revision.

File: backend/lcfs/web/api/transaction/repo.py [hunk 2]
```diff
@@ -67,33 +67,26 @@ async def get_transactions_paginated(self, offset: int, limit: int, conditions:
...
         else:
-            # Fetch visible statuses for government, but only for "Transfer" transactions
-            visible_statuses_government = await self.get_visible_statuses(EntityType.Government)
-            government_condition = and_(
+            # Government view should see all non-transfer transactions regardless of status
+            gov_transfer_condition = and_(
                 transfer_type_condition,
-                TransactionView.status.in_(
-                    [status.value for status in visible_statuses_government])
+                TransactionView.status.in_([status.value for status in await self.get_visible_statuses(EntityType.Government)])
```
Comment (prv-proton) [inline backend/lcfs/web/api/transaction/repo.py:85]: Combining statuses is causing duplicates.
Transaction statuses in the filter drop down of ag-grid is showing duplicate values.

File: frontend/src/views/Transactions/components/TransactionView.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { useTranslation } from 'react-i18next';
+import BCBox from '@/components/BCBox';
+import BCTypography from '@/components/BCTypography';
+import { Grid } from '@mui/material';
+import { LabelBox } from './LabelBox'
+
+// Define common inline styles
+const inlineLabelStyle = { display: 'inline', marginRight: 6 };
+
+export const TransactionView = ({ transaction }) => {
```
Comment (prv-proton) [inline frontend/src/views/Transactions/components/TransactionView.jsx:11]: Breadcrumb link for "Admin adjustment" isn't working
image

**Comment by prv-proton** [comment]
When I'm on analyst role, the transaction history isn't available to view. Is this expected?
image

File: frontend/src/assets/locales/en/transaction.json [hunk 4]
```diff
@@ -35,16 +35,29 @@
...
   "effectiveDate": "<b>Effective date</b> (optional, if blank will be effective on the date of director approval):",
+  "effectiveDateLabel": "Effective date:",
   "complianceUnits": "<b>Compliance units</b> (negative or positive amounts for administrative adjustment):",
+  "complianceUnitsLabel": "Compliance units:",
+  "comments": "<b>Comment</b> (can be edited below):",
   "initiativeAgreement": "Initiative agreement",
   "administrativeAdjustment": "Administrative adjustment",
   "organization": "Organization:",
   "selectOrgLabel": "Select organization",
   "complianceBalance": "Compliance Balance:",
```
Comment (prv-proton) [inline frontend/src/assets/locales/en/transaction.json:53]: duplicated field.

File: backend/lcfs/web/api/admin_adjustment/schema.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+from lcfs.web.api.base import BaseSchema
+from typing import Optional, List
+from datetime import date, datetime
+
+class AdminAdjustmentStatusSchema(BaseSchema):
+    admin_adjustment_status_id: int
+    status: str
+
+    class Config:
```
Comment (prv-proton) [inline backend/lcfs/web/api/admin_adjustment/schema.py:9]: this code might not be required, as we're already adding it in BaseSchema and also with latest pydantic version, they've have changed the way to add this in dict format.

File: frontend/src/views/Transactions/Transactions.jsx [hunk 6]
```diff
@@ -15,8 +15,8 @@ import { useTranslation } from 'react-i18next'
 import { Role } from '@/components/Role'
 import { transactionsColDefs } from './_schema'
 import { useCurrentUser } from '@/hooks/useCurrentUser'
+import { ORGANIZATION_STATUSES, TRANSACTION_STATUSES, TRANSFER_STATUSES } from '@/constants/statuses'
 import { roles, govRoles } from '@/constants/roles'
-import { ORGANIZATION_STATUSES, TRANSFER_STATUSES } from '@/constants/statuses'
 import OrganizationList from './components/OrganizationList'
```
Comment (prv-proton) [inline frontend/src/views/Transactions/Transactions.jsx:20]: Due to slow connection, I tried to click more than once on "Recommend" button on the confirmation Modal window and it ended up creating multiple records.
image

File: frontend/src/views/Transactions/AddEditViewTransaction.jsx [hunk 7]
```diff
@@ -15,137 +17,300 @@ import {
...
-      transactionId,
-      methods,
-      t
-    ]
+    [transactionId, txnType, methods, t, setModalData, createUpdateAdminAdjustment, createUpdateInitiativeAgreement, hasRoles]
   )
 
-  return <>
+  if (transactionId && isTransactionDataLoading)
+    return <Loading message={t(`${txnType}:loadingText`)} />
```
Comment (prv-proton) [inline frontend/src/views/Transactions/AddEditViewTransaction.jsx:205]: txnType will be null here, hence loadingText isn't loading the intented text.

**Review by prv-proton — CHANGES_REQUESTED** [review]
Hi Alex, code looks good, have provided minor issues that might need to be addressed. Thanks.

File: frontend/src/views/Transactions/Transactions.jsx (hunk 6 above)
Comment (AlexZorkin) [reply frontend/src/views/Transactions/Transactions.jsx:20]: I added a loading indicator to the BCButton and the Modal triggers it on button click now. This prevents the double clicking.

File: frontend/src/views/Transactions/components/TransactionView.jsx (hunk 3 above)
Comment (AlexZorkin) [reply frontend/src/views/Transactions/components/TransactionView.jsx:11]: I updated our Crumb.jsx component to better handle custom breadcrumbs. 

```Jsx
  // Mapping for custom breadcrumb labels and routes
  const customBreadcrumbs = {
    'admin': { label: 'Administration', route: '/admin' },
    'transfers': { label: 'Transactions', route: '/transactions' },
    'add-org': { label: 'Add organization', route: '/add-org' },
    'edit-org': { label: 'Edit organization', route: '/edit-org' },
    'initiative-agreement': { label: 'Transactions', route: '/transactions' },
    'admin-adjustment': { label: 'Transactions', route: '/transactions' }
  }
  
  const isLast = index === pathnames.length - 1
          const customCrumb = customBreadcrumbs[name] || {}
          const routeTo = customCrumb.route || `/${pathnames.slice(0, index + 1).join('/')}`
          const displayName = customCrumb.label || name.charAt(0).toUpperCase() + name.slice(1).replace('-', ' ')

**Pull request merged**